### PR TITLE
PR Webhook - Add base/head info

### PR DESCRIPTION
**Pull request by kmadel**
Regarding https://github.com/gogits/gogs/issues/3484

Requires https://github.com/gogits/go-gogs-client/pull/47

### CODE REVIEW

File: models/pull.go [hunk 1]
```diff
@@ -114,18 +114,22 @@ func (pr *PullRequest) LoadIssue() (err error) {
...
+		Milestone:  apiIssue.Milestone,
+		Assignee:   apiIssue.Assignee,
+		State:      apiIssue.State,
+		Comments:   apiIssue.Comments,
+		HTMLURL:    pr.Issue.HTMLURL(),
+		HasMerged:  pr.HasMerged,
+		HeadBranch: pr.HeadBranch,
+		HeadRepo:   pr.HeadRepo.APIFormat(nil),
+		BaseBranch: pr.BaseBranch,
+		BaseRepo:   pr.BaseRepo.APIFormat(nil),
```
Comment (Unknwon) [inline models/pull.go]: See https://www.notion.so/Coding-Guidelines-15a29828afc34409b50e030a20120722

File: models/pull.go [hunk 2]
```diff
@@ -114,18 +114,22 @@ func (pr *PullRequest) LoadIssue() (err error) {
...
+		Poster:     apiIssue.Poster,
+		Title:      apiIssue.Title,
+		Body:       apiIssue.Body,
+		Labels:     apiIssue.Labels,
+		Milestone:  apiIssue.Milestone,
+		Assignee:   apiIssue.Assignee,
+		State:      apiIssue.State,
+		Comments:   apiIssue.Comments,
+		HeadBranch: pr.HeadBranch,
+		HeadRepo:   pr.HeadRepo.APIFormat(nil),
```
Comment (Unknwon) [inline models/pull.go]: Have you check all the places that calls `PullRequest.APIFormat` ensure `HeadRepo` and `BaseRepo` are not nil?

Comment (kmadel) [inline models/pull.go]: Will double check - OR would it be better to do a nil check here and return nil?

Comment (Unknwon) [inline models/pull.go]: See https://github.com/kmadel/gogs/blob/324327eee031d92afdadde0da7b65ca1f095acd3/models/pull.go#L111 and update it

**Comment by lstahlman** [comment]
Any update on this PR?

**Comment by marcinwyszynski** [comment]
It would be fantastic to have this data. I'm trying to add Gogs support to [codebeat](https://codebeat.co) and we're missing crucial pieces of info in the webhook payload:
- source ref;
- source repo;
- base ref;
- base repo;
- last source commit hash;

GitHub, Bitbucket and Gitlab all provide these fields on one form or another so they're pretty much an industry standard and enable cool integrations.

**Comment by hhenkel** [comment]
@kmadel Would it be possible for you to open this PR also on https://github.com/go-gitea/gitea ?

**Comment by kmadel** [comment]
@hhenkel sure, np

File: models/pull.go (hunk 2 above)
Comment (kmadel) [inline models/pull.go]: Added nil checks and comments

**Comment by Unknwon** [comment]
Thanks!

**Pull request merged**